src/fileio/datafile.cxx

```diff
@@ -88,12 +88,14 @@ Datafile& Datafile::operator=(const Datafile &rhs) {
   f3d_arr      = rhs.f3d_arr;
   v2d_arr      = rhs.v2d_arr;
   v3d_arr      = rhs.v3d_arr;
-  filenamelen=FILENAMELEN;
-  filename     = new char[filenamelen];
   return *this;
 }
 
 Datafile::~Datafile() {
+  if (file){
+    delete file;
+    file = NULL;
```

Comment:
`nullptr` is preferred to `NULL`, because `NULL` can be implicitly cast to `int`, which can cause all sorts of headaches. `nullptr` is its own type, which is comparable with all pointer types.

We probably should change `NULL` to `nullptr` everywhere